### ISSUE TITLE
Bug in generated opm-common-config.cmake?

### ISSUE THREAD

**Issue by joakim-hove** [OPM/opm-common#456]
The situation here is that I want to compile the `sunbeam` python wrapper/extension to the parser functionality in opm: https://github.com/Statoil/sunbeam/

This previously linked to the opm-parser project; I now want to bring it into 2018 and link to the opm-common project: https://github.com/Statoil/sunbeam/pull/85

The opm-common project has been built and installed, and in sunbeam I am looking for an installed version of opm-common:

```cmake
find_package(opm-common REQUIRED)
```
This configures and compiles nicely, but at link time the project fails with:
```
Linking CXX shared module ../python/sunbeam/libsunbeam.so
/usr/bin/ld: cannot find -lecl
```

Looking at the installed `opm-common-config.cmake` it contains this line:
```cmake
set (opm-common_LIBRARIES ${opm-common_LIBRARY} "ecl;/private/joaho/work/ERT/install/lib64/libecl.so.2.3;m;dl;-lpthread;/usr/lib64/libz.so;/usr/lib64/libboost_system-mt.so;/usr/lib64/libboost_filesystem-mt.so;/usr/lib64/libboost_regex-mt.so")
  mark_as_advanced (opm-common_LIBRARY)
```
i.e. the ecl library is mentioned twice - first as just `ecl` and then subsequently with full path. If I remove the `ecl` element from the list the linking of sunbeam succeeds. The multiple mention of `ecl` looked a bit suspicious to me - is it a bug?

**Comment by bska** [comment]
> The multiple mention of ecl looked a bit suspicious to me - is it a bug?

This may (or may not) be related to #339.

**Comment by blattms** [comment]
@bska That patch is included in 2018.04

@joakim-hove I wanted to replicate this, but I cannot even run Cmake with your branch of sunbeam as I am getting errors:
```
-- Found PythonInterp: /usr/bin/python (found version "2.7.13") 
CMake Error at CMakeLists.txt:26 (add_subdirectory):
  The source directory

    /tmp/sunbeam/pybind11

  does not contain a CMakeLists.txt file.


CMake Error at python/CMakeLists.txt:1 (include):
  include could not find load file:

    FindPythonModule


CMake Error at python/CMakeLists.txt:2 (include):
  include could not find load file:

    PythonPackage


CMake Error at python/CMakeLists.txt:4 (python_module):
  Unknown CMake command "python_module".


-- Configuring incomplete, errors occurred!
```
pybind11 is an empty directory. Neither FindPythonModule.cmake nor PythonPackage are available on my system.

My shot at this is that the quotation marks in opm_common_LIBRARIES variable could be the problem as they may prevent substitution of ecl with the actual library of that target. Can you try this patch:
```diff
diff --git a/cmake/Templates/opm-project-config.cmake.in b/cmake/Templates/opm-project-config.cmake.in
index 421708a8..e2b700dd 100644
--- a/cmake/Templates/opm-project-config.cmake.in
+++ b/cmake/Templates/opm-project-config.cmake.in
@@ -34,7 +34,7 @@ if(NOT @opm-project_NAME@_FOUND)
 
   # libraries come from the build tree where this file was generated
   set (@opm-project_NAME@_LIBRARY "@opm-project_LIBRARY@")
-  set (@opm-project_NAME@_LIBRARIES ${@opm-project_NAME@_LIBRARY} "@opm-project_LIBRARIES@")
+  set (@opm-project_NAME@_LIBRARIES ${@opm-project_NAME@_LIBRARY} @opm-project_LIBRARIES@)
   mark_as_advanced (@opm-project_NAME@_LIBRARY)
 
   # not all projects have targets; conditionally add this part
```

**Comment by blattms** [comment]
Thanks, recursive cloning did the trick. My guess was wrong.

**Comment by blattms** [comment]
The bug was in OpmFind.cmake. Fixed it in #457

**Comment by blattms** [comment]
@joakim-hove My fix might be wrong and will not help you with the release. A simple workaround would be to add a line `find_package(ecl REQUIRED)` before searching for opm_common.  Then the target ecl is defined and resolved.

**Comment by joakim-hove** [comment]
Fixed in #459

**Comment by blattms** [comment]
thanks for the feedback
